File: 2023/day4/part1.nim
import std/math
import std/sequtils
import std/strutils

proc get_card_points*(line: string): int = 
  var numbers = line.split(':')[1]
  var split_numbers = numbers.split('|')
  var winning_numbers = split_numbers[0].splitWhitespace().map(parseInt)
  var mynumbers = split_numbers[1].splitWhitespace().map(parseInt)
  mynumbers.filterIt(it in winning_numbers).len()

proc main() =
  var tot = 0
  for line in lines("input_small.txt"):
    var points = get_card_points(line)
    if points > 0:
      tot += 2^(points - 1)
  echo tot

when isMainModule:
  main()

File: 2023/day4/part1.py
def get_card_points(card: str):
  _, numbers = card.split(': ')
  winning_text, mytext = numbers.split(' | ')
  winning_numbers = [int(v) for v in winning_text.split()]
  mynumbers = [int(v) for v in mytext.split() if int(v) in winning_numbers]
  return len(mynumbers)


def main():
  tot = 0
  with open('input_small.txt') as f:
    for line in f:
      points = get_card_points(line.strip())
      if points > 0:
        tot += pow(2, points - 1)
  print(tot)


if __name__ == '__main__':
  main()